src/Xamarin.Android.Build.Tasks/Utilities/ResourceDesignerImportGenerator.cs

```diff
@@ -34,52 +36,71 @@ public ResourceDesignerImportGenerator (string ns, CodeTypeDeclaration applicati
 			}
 		}
 
-		public void CreateImportMethods (IEnumerable<AssemblyDefinition> libraries)
+		public void CreateImportMethods (IEnumerable<string> libraries)
 		{
 			var method = new CodeMemberMethod () { Name = "UpdateIdValues", Attributes = MemberAttributes.Public | MemberAttributes.Static };
 			primary.Members.Add (method);
-			foreach (var assembly in libraries) {
-				var att = assembly.CustomAttributes.FirstOrDefault (
-					ca => ca.AttributeType.FullName == "Android.Runtime.ResourceDesignerAttribute");
-				if (att == null)
-					continue;
-				if ((bool) att.Properties.First (p => p.Name == "IsApplication").Argument.Value == true)
-					continue; // application resource IDs are constants, cannot merge.
-				var td = assembly.Modules.SelectMany (m => m.Types).FirstOrDefault (
-					t => t.FullName == (string)att.ConstructorArguments.First ().Value);
-
-				// F# has no nested types, so we need special care.
-				if (td.NestedTypes.Any ())
-					CreateImportFor (true, td.NestedTypes, method);
```

Comment:
TODO, I need to figure out the exact state of F# here. Do F# projects even call this or do they use this?

https://www.nuget.org/packages/Xamarin.Android.FSharp.ResourceProvider/